### PR TITLE
[CARBONDATA-1155] Load query were failing if sort_columns have provided nodictionary dimension

### PR DESCRIPTION
**Pull request by rahulforallp**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2392/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/271/

**Comment by rahulforallp** [comment]
retest this please

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/sortcolumns/TestSortColumns.scala [hunk 1]
```diff
@@ -37,6 +37,12 @@ class TestSortColumns extends QueryTest with BeforeAndAfterAll {
     checkAnswer(sql("select empno from sorttable1"), sql("select empno from sorttable1 order by empno"))
   }
 
+  test("create table with no dictionary sort_columns with dictionary exclude") {
```
Comment (QiangCai) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/sortcolumns/TestSortColumns.scala]: Suggest to add test case for:
NumberOfNoDictSortColumns < NoDictionaryCount
It means  some no dictionary columns are in sort_columns and some no dictioanry columns are not.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2395/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/274/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/282/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/282/org.apache.carbondata$carbondata-spark-common-test/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-spark-common-test</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/282/org.apache.carbondata$carbondata-spark-common-test/testReport/org.apache.carbondata.spark.testsuite.sortcolumns/TestSortColumns/create_table_with_no_dictionary_sort_columns_where_NumberOfNoDictSortColumns___NoDictionaryCount/'><strong>org.apache.carbondata.spark.testsuite.sortcolumns.TestSortColumns.create table with no dictionary sort_columns where NumberOfNoDictSortColumns < NoDictionaryCount</strong></a></li></ul>

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2402/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/291/

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/sortcolumns/TestSortColumns.scala (hunk 1 above)
Comment (rahulforallp) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/sortcolumns/TestSortColumns.scala]: @QiangCai  i have added test case for NumberOfNoDictSortColumns < NoDictionaryCount

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2412/

**Comment by QiangCai** [comment]
please rebase to resolve confilct

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2433/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/314/

**Comment by QiangCai** [comment]
LGTM

**Comment by QiangCai** [comment]
@jackylk please review also

File: processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java [hunk 2]
```diff
@@ -603,7 +603,7 @@ private void copyLocalFileToCarbonStore(String carbonStoreFilePath, String local
     short numberOfByteToStorLength = 2;
     // as length part is removed, so each no dictionary value index
     // needs to be reshuffled by 2 bytes
-    for (int i = 0; i < dataWriterVo.getNoDictionaryCount(); i++) {
+    for (int i = 0; i < dataWriterVo.getSegmentProperties().getNumberOfNoDictSortColumns(); i++) {
```
Comment (jackylk) [inline processing/src/main/java/org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java]: Please create a local variable for `dataWriterVo.getSegmentProperties().getNumberOfNoDictSortColumns()`, it is better not put in loop

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2480/

**Comment by rahulforallp** [comment]
@jackylk code has refactored. Please review it.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/365/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**